taqueria-vscode-extension/src/lib/helpers.ts

```diff
@@ -328,6 +330,46 @@ export const inject = (deps: InjectedDependencies) => {
 			);
 		};
 
+	const updateCommandStates = async (
+		context: api.ExtensionContext,
+		output: api.OutputChannel,
+		i18n: i18n,
+		projectDir: Util.PathToDir,
+	) => {
+		try {
+			const config = await Util.TaqifiedDir.create(projectDir, i18n);
+			vscode.commands.executeCommand('setContext', '@taqueria-state/is-taqified', !!config.config);
+			const plugins = getWellKnownPlugins();
+			for (const plugin of plugins) {
+				const found = config.config.plugins?.find(item => item.name === plugin) !== undefined;
+				vscode.commands.executeCommand('setContext', plugin, found);
+			}
+		} catch (e: any) {
+			// After logging PR is merged, log this error
+		}
+	};
+
+	const createWatcherIfNotExists = (
+		context: api.ExtensionContext,
+		output: api.OutputChannel,
+		i18n: i18n,
+		projectDir: Util.PathToDir,
+		createConfigWatcherIfNotExists: (folder: string, factory: () => api.FileSystemWatcher) => void,
+	) => {
+		createConfigWatcherIfNotExists(projectDir, () => {
+			const watcher = vscode.workspace.createFileSystemWatcher(join(projectDir, '.taq/config.json'));
+			// TODO: We should detect the event that VsCode's current Folder is changed and the watcher should be disposed
+
+			updateCommandStates(context, output, i18n, projectDir);
+
+			// TODO: Is passing these arguments to the callback of a long lived watcher prevent GC? Are these short lived objects?
```

Comment:
Indeed, as they're in closure scope, they will be long lived. 

As an alternative, I'd say adjust updateCommandStates to accept an object of key/pair values instead:
```
updateCommandStates({
  context,
  output,
  i18n,
  projectDir
})
```

Saying that, many of those will just be references so its probably not a significant issue.